From 5ff7c197d871c0ac0514df0a07ae43c262a7c11f Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Mon, 18 Apr 2011 16:18:35 +0800
Subject: [PATCH 0117/2463] ENGR00142266-6 eSDHC:Add the GPIO CD/WP on MX53
 ARD boards

arm: enable the eSDHC CD/WP on MX53 ARD board

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 arch/arm/mach-mx5/board-mx53_ard.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-mx5/board-mx53_ard.c b/arch/arm/mach-mx5/board-mx53_ard.c
index b200b98e..070c26f 100755
--- a/arch/arm/mach-mx5/board-mx53_ard.c
+++ b/arch/arm/mach-mx5/board-mx53_ard.c
@@ -36,7 +36,9 @@
 #include "crm_regs.h"
 #include "devices-imx53.h"
 
+#define ARD_SD1_CD IMX_GPIO_NR(1, 1)
 #define ARD_SD1_WP IMX_GPIO_NR(1, 9)
+#define ARD_SD2_CD IMX_GPIO_NR(1, 4)
 #define ARD_SD2_WP IMX_GPIO_NR(1, 2)
 #define ARD_TS_INT IMX_GPIO_NR(7, 12)
 #define ARD_ETHERNET_INT_B IMX_GPIO_NR(2, 31)
@@ -181,11 +183,13 @@ static const struct imxuart_platform_data mx53_ard_uart_data __initconst = {
 	.flags = IMXUART_HAVE_RTSCTS,
 };
 
-static const struct esdhc_platform_data mx53_ard_sd0_data __initconst = {
+static const struct esdhc_platform_data mx53_ard_sd1_data __initconst = {
+	.cd_gpio = ARD_SD1_CD,
 	.wp_gpio = ARD_SD1_WP,
 };
 
-static const struct esdhc_platform_data mx53_ard_sd1_data __initconst = {
+static const struct esdhc_platform_data mx53_ard_sd2_data __initconst = {
+	.cd_gpio = ARD_SD2_CD,
 	.wp_gpio = ARD_SD2_WP,
 };
 
@@ -227,8 +231,8 @@ static void __init mx53_ard_board_init(void)
 					ARRAY_SIZE(mx53_ard_pads));
 	mx53_ard_init_uart();
 	imx53_add_imx2_wdt(0, NULL);
-	imx53_add_sdhci_esdhc_imx(0, &mx53_ard_sd0_data);
-	imx53_add_sdhci_esdhc_imx(1, &mx53_ard_sd1_data);
+	imx53_add_sdhci_esdhc_imx(0, &mx53_ard_sd1_data);
+	imx53_add_sdhci_esdhc_imx(1, &mx53_ard_sd2_data);
 
 	weim_cs_config();
 	mx53_ard_io_init();
-- 
1.7.7.4

